src/ur5e_arm.cpp

```diff
@@ -403,6 +403,54 @@ void UR5eArm::send_trajectory(const std::vector<vector6d_t>& p_p,
 
 // helper function to read a data packet and send a noop message
 void UR5eArm::read_and_noop() {
+    // check to see if an estop has occurred.
+    std::string status;
+    dashboard->commandSafetyStatus(status);
+
+    if (status.find(urcl::safetyStatusString(urcl::SafetyStatus::NORMAL)) == std::string::npos) {
+        // the arm is currently estopped. save this state.
+        estop.store(true);
+    } else {
+        // the arm is in a normal state.
+        if (estop.load()) {
+            // if the arm was previously estopped, attempt to recover from the estop.
+            // We should not enter this code without the user interacting with the arm in some way(i.e. resetting the estop)
```

Comment:
How do we determine if this is the case?

Reply:
we know the user has switched off the estop when the `commandSafetyStatus(status)` returns `NORMAL`, and if we had already triggered a non-NORMAL status before.